app.rb

```diff
@@ -1,7 +1,34 @@
 require 'sinatra/base'
+require 'sinatra/reloader' 
+require_relative 'lib/player.rb'
+require_relative 'lib/game.rb'
+require_relative 'lib/weapon.rb'
+require_relative 'lib/computer.rb'
+
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+  enable :sessions
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  before do
+    @game = Game.instance
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  post '/names' do
+    player_1 = Player.new(params[:player_1_name])
+    player_weapon = Weapon.new(params[:weapon])
+    computer_weapon = Computer.new.random_weapon
+    @game = Game.create(player_1, player_weapon, computer_weapon)
+    redirect '/result'
+  end
+
+  get '/result' do
+    erb @game.result
```

Comment:
It might be worth double checking the return value of `@game.result` before calling the view - in case the method returns something else than `lose`, `win` or `draw` (for whatever reason), the app will crash as there's no such view!